bamboo/clean.sh

```diff
@@ -6,19 +6,18 @@ LBANN_DIR=$(git rev-parse --show-toplevel)
 # Compiler Tests
```

Comment:
Might be good to add a comment up at line 3 like `# Bamboo has an option to clean the directory after building, but that would remove log files that we may want to look at in between builds, so we need to clean the directory before building`. Otherwise, I could see someone one day thinking this `clean.sh` script isn't needed.